url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/ccproxy/AgentConnection.java

```diff
@@ -0,0 +1,1042 @@
+					return false;
+			}
+
+            String querystr="select va.nazwisko, va.numer, va.supervisor, tu.haslo as md5pass " +
+                    "from v_agent va join t_user tu on va.id_user=tu.id_user where tu.login='"
+                    + agentNumber + "'";
+            if (offline){
+                querystr="select va.nazwisko, va.numer, va.supervisor, tu.haslo as md5pass " +
+                        "from v_agent va join t_user tu on va.id_user=tu.id_user where va.numer='"
+                        + agentNumber + "'";
+            }
+			rs = p.dbConn
+				.query(querystr);
+			if ((rs != null) && rs.next()) {
+				if (p.c.externAuthURL==null){
+					String md5Pass = new String(rs.getString("md5pass"));
+					if (!offline && !md5Pass.equals(MD5Crypt.crypt(agentPassword, md5Pass)))
+						return false;
+				} 
+			} else return false;
+		} else {
+			//rs = p.dbConn
+			//	.query("select nazwisko, numer, supervisor from v_agent where numer='"
+			//			+ agentNumber + "' and haslo='" + agentPassword + "'");
+			//if ((rs != null) && rs.next()) {
+			//	
+			//} else return false;
+			return false; //nie ma juz autoryzacji "agentem"
+		}
+		
+		{
+				agentName = new String(rs.getString("nazwisko"));
+				agentNumber = new String(rs.getString("numer"));
+				if (rs.getString("supervisor").equals("t"))
+					supervisor = true;
+				
+				if (agentExten.length()==0){
+					rs = p.dbConn
+					.query("select vus.numer from v_agent va join v_user_sip vus on va.id_user=vus.id_user where va.numer='"
+							+ agentNumber + "'");
+					if ((rs != null) && rs.next()){
+						agentExten = new String(rs.getString("numer"));
+					} else return false;
+				}
+				
+				out.println("Auth success.");
+				setThreadName();
+				p.log.info("Database auth success.");
+				agentMakeQueuePenalty(agentNumber);
+				return true;
+			}
+		} catch (SQLException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		} catch (NoSuchAlgorithmException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		out.println("+OK autoryzacja w bazie danych i po niej juz.");
+
+		return false;
+
+	}
+
+	private boolean externAuth(String url, String agentName, String password){
+		boolean authorized=false;
+		try {
+			HttpURLConnection con = (HttpURLConnection) new URL(url).openConnection();
+			con.setRequestMethod("POST");
+			con.setDoOutput(true);
+			con.getOutputStream().write(("user="+agentName).getBytes("UTF-8"));
+			con.getOutputStream().write("&".getBytes("UTF-8"));
+			con.getOutputStream().write(("pass="+password).getBytes("UTF-8"));
+			//con.setRequestMethod("GET");
+			BufferedReader br = new BufferedReader(new InputStreamReader(con.getInputStream()));
+			String inputLine;
+			while ((inputLine = br.readLine()) != null){
+				if ("OK".equals(inputLine)) 
+					authorized=true;
+			}
+			br.close();
+		} catch (MalformedURLException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This code contains a potential information leakage in the handling of the following *exception logging*: (MalformedURLException e). Stack trace information reveals variable names, method names, line numbers, and other parts of the code.

**Fix**
Release data more selectively, or declassify the data before it&#39;s released. 

**More info**
[Learn more about information leakage](https://cwe.mitre.org/data/definitions/497.html) (external link).